packages/table/src/table.tsx

```diff
@@ -615,6 +622,31 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         }
     }
 
+    private maybeScrollTableIntoView() {
+        const { viewportRect } = this.state;
+
+        const maxRowIndex = this.grid.numRows - 1;
+        const maxColIndex = this.grid.numCols - 1;
+
+        const tableBottom = this.grid.getCumulativeHeightAt(maxRowIndex);
+        const tableRight = this.grid.getCumulativeWidthAt(maxColIndex);
+
+        let nextScrollLeft = viewportRect.left;
+        let nextScrollTop = viewportRect.top;
+
+        if (tableBottom < viewportRect.top + viewportRect.height) {
+            // scroll the last row into view
+            nextScrollTop = Math.max(0, tableBottom - viewportRect.height);
+        }
+
+        if (tableRight < viewportRect.left + viewportRect.width) {
+            // scroll the last column into view
+            nextScrollLeft = Math.max(0, tableRight - viewportRect.width);
+        }
```

Comment:
these could be done as ternaries, rather than `let`s:

```js
const nextScrollLeft = (tableRight < viewportRect.left + viewportRect.width)
    ? Math.max(0, tableRight - viewportRect.width)
    : viewportRect.left;
```